autoload/ale/highlight.vim

```diff
@@ -26,6 +26,24 @@ endif
 let s:MAX_POS_VALUES = 8
 let s:MAX_COL_SIZE = 1073741824 " pow(2, 30)
 
+" Check if we have neovim's buffer highlight API
+"
+" Below we define some functions' implementation conditionally if this API
+" exists or not.
+"
+" The API itself is more ergonomic and neovim performs highlights positions
+" rebases during edits so we see less stalled highlights.
+let s:nvim_api = exists('*nvim_buf_add_highlight') && exists('*nvim_buf_clear_namespace')
+
+function! s:ale_nvim_highlight_id(bufnr) abort
+    if get(b:, 'ale_nvim_highlight_id', -1) == -1
```

Comment:
Use `getbufvar` instead of `b:` variables. The variables for `b:` can differ from from buffer for the given buffer number.